windows/RNPrint/RNPrint.h

```diff
@@ -0,0 +1,402 @@
+#pragma once
+
+#include "pch.h"
+#include "NativeModules.h"
+
+#include <functional>
+#include <sstream>
+#include <mutex>
+
+#include "JSValue.h"
+
+using namespace winrt::Microsoft::ReactNative;
+using namespace winrt::Windows::Data::Pdf;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::Storage::Streams;
+using namespace winrt::Windows::Web::Http;
+using namespace winrt::Windows::Graphics::Printing;
+using namespace winrt::Windows::Graphics::Printing::OptionDetails;
+using namespace winrt::Windows::UI::Xaml::Printing;
+using namespace winrt::Windows::UI::Xaml::Controls;
+using namespace winrt::Windows::UI::Xaml::Media::Imaging;
+using namespace winrt::Windows::UI::Xaml::Media;
+
+
+namespace winrt::RNPrint
+{
+  // Represent the options object passed to print()
+  struct RNPrintOptions
+  {
+    std::optional<std::string> html = std::nullopt;
+    std::optional<std::string> filePath = std::nullopt;
+    bool isLandscape = false;
+    std::string jobName;
+  };
+
+  REACT_MODULE(RNPrint);
+  struct RNPrint
+  {
+    const std::string Name = "RNPrint";
+
+    // Default name for the print job.
+    inline static const std::string defaultJobName = "Document";
+
+    // Name for the XAML Canvas to add preview folders to.
+    inline static const std::string RNPrintCanvasName = "RNPrintCanvas";
```

Comment:
these can be `constexpr` instead

Reply:
Will change them to `char[]` as well, so that they can be `constexpr`.

Reply:
std::string can be constexpr with c++20, or you can use string_view. In general try avoiding C style character strings/arrays